addons/website_sale/models/product_template.py

```diff
@@ -273,6 +273,12 @@ def _get_combination_info(self, combination=False, product_id=False, add_qty=1,
                     fields.Date.today())
             has_discounted_price = pricelist.currency_id.compare_amounts(list_price, price) == 1
             prevent_zero_price_sale = not price and current_website.prevent_zero_price_sale
+
+            compare_list_price = self.compare_list_price
+            if self.currency_id != pricelist.currency_id:
```

Comment:
```suggestion
            if pricelist and pricelist.currency_id != self.currency_id:
```
Everything can now work without any pricelist on every sale module. eCommerce requires pricelists once we are in a multi-currency settings though, so if there is a risk of different currency, then there must be a pricelist.